File: frontend/src/models/Student.js
export class StudentResponse {
  constructor({id,code, name, gender, dob, className, phone, email, major, faculty, role
  }) {
    this.id = id
    this.code = code
    this.name = name
    this.gender = gender
    this.dob = dob
    this.className = className
    this.phone = phone
    this.email = email
    this.major = major
    this.faculty = faculty
    this.role = role
  }

  static fromJSON(json) {
    return new StudentResponse(json)
  }
}

File: backend/src/main/java/com/examreg/examreg/models/Student.java
package com.examreg.examreg.models;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.List;

import com.examreg.examreg.enums.Gender;
import com.examreg.examreg.enums.Role;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.OneToMany;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.Setter;

@AllArgsConstructor
@Getter
@Setter
@Entity
public class Student extends User{

  public Student() {
    this.setRole(Role.STUDENT);
  }

  @Column(nullable = false)
  private String studentCode;

  @Column(nullable = false)
  private String fullname;

  @Enumerated(EnumType.STRING)
  private Gender gender;

  private String className;
  
  private String phone;

  private String major;

  private LocalDate dob;

  private String faculty;

  private boolean firstLogin = true;

  private LocalDateTime loginLockedUntil;

  @OneToMany(mappedBy = "student", cascade = CascadeType.ALL, orphanRemoval = true)
  private List<RegistrationSession> registrationSessions;

  @OneToMany(mappedBy = "student")
  private List<StudentSubjectStatus> studentSubjects;

  @OneToMany(mappedBy = "student")
  private List<ExamRegistration> examRegistrations;
}
